### PR TITLE
build: update to Aspect CLI 5.5.2

### PR DESCRIPTION
**Pull request by gregmagolan**
Required adding a few import ignore directives in `client/web/BUILD.bazel`. @jbedard is working on a fix for those. Has to do with tsconfig paths. Fix will land in a future release.

## Test plan

5.4.11 output:

```
$ bazel configure
Updating BUILD files for protobuf, go, javascript
WARN[0000] Failed to load base tsconfig file @sourcegraph/tsconfig: open /Users/greg/aspect/sourcegraph/sourcegraph/@sourcegraph/tsconfig: no such file or directory 
client/app-shell/src/app-shell.tsx parse error(s):
50: Top-level await is currently not supported with the "iife" output format
Failed to parse tsconfig file client/completions-review-tool/tsconfig.json: ERR: position:298 object key must be string pos:298 : 
933 BUILD files visited
0 BUILD files updated
```

5.5.2 output has some new spam as well:

```
$ bazel configure
Updating BUILD files for protobuf, go, javascript
WARN[0000] Failed to load base tsconfig file @sourcegraph/tsconfig: open /Users/greg/aspect/sourcegraph/sourcegraph/@sourcegraph/tsconfig: no such file or directory 
Failed to parse tsconfig file client/completions-review-tool/tsconfig.json: ERR: position:298 object key must be string pos:298 : 
client/web/src/globals.d.ts:
	     17: }
         ^

client/web/src/storm/backend/route-loader.ts:
	     41:     usePreloadedQueryData: () => ReturnType<typeof useSuspenseQuery<D, V>>
                                                    ^
	     41:     usePreloadedQueryData: () => ReturnType<typeof useSuspenseQuery<D, V>>
                                                     ^

934 BUILD files visited
0 BUILD files updated
```

### CODE REVIEW

File: client/shared/src/testing/BUILD.bazel [hunk 1]
```diff
@@ -99,18 +97,36 @@ ts_project(
...
+        "//client/shared:graphql_operations",
         "//client/shared:node_modules/@sourcegraph/build-config",
         "//client/shared:node_modules/@sourcegraph/client-api",
         "//client/shared:node_modules/@sourcegraph/common",
         "//client/shared:node_modules/@sourcegraph/http-client",
         "//client/shared:shared_lib",
     ],
 )
+
+ts_project(
```
Comment (jbedard) [inline client/shared/src/testing/BUILD.bazel:116]: I think this is wrong as well. Again most likely something to do with priority of globs and how they are inherited/overridden.

File: client/web/dev/BUILD.bazel [hunk 2]
```diff
@@ -20,7 +20,6 @@ ts_config(
 ts_project(
     name = "dev",
     srcs = [
-        "mocks/mockEventLogger.ts",
```
Comment (jbedard) [inline client/web/dev/BUILD.bazel:23]: Same here, the glob should have been overridden and everything should be tests.

Comment (gregmagolan) [reply client/web/dev/BUILD.bazel:23]: Fixed with 5.5.1

File: client/shared/src/testing/BUILD.bazel (hunk 1 above)
Comment (gregmagolan) [reply client/shared/src/testing/BUILD.bazel:116]: Fixed with 5.5.1

Comment (gregmagolan) [reply client/shared/src/testing/BUILD.bazel]: Fixed with update to 5.5.2

File: client/web/dev/BUILD.bazel (hunk 2 above)
Comment (gregmagolan) [reply client/web/dev/BUILD.bazel]: Fixed with update to 5.5.2

**Review by jhchabran — APPROVED** [review]

**Comment by valerybugakov** [comment]
Hey @gregmagolan, is there anything we can do to fix/hide these spam messages?

**Comment by gregmagolan** [comment]
> Hey @gregmagolan, is there anything we can do to fix/hide these spam messages?

They are coming out of treesitter (which is used by gazelle plugin to parse the typescript files). @jbedard Is planning to suppress the tree-sitter AST parser warnings so they'll go away in an Aspect CLI release in the near future.

File: client/storybook/BUILD.bazel [hunk 3]
```diff
@@ -92,7 +91,6 @@ ts_project(
     testonly = True,
```
Comment (rc-glean) [inline client/storybook/BUILD.bazel:91]: Apologies if this is a really dumb question, but is sourcegraph tooled to support a bazel test target for executing storybook tests?

I didn't find any dependencies of the storybook_tests ts_project with this codesearch https://github.com/search?type=code&q=%22storybook_tests%22+org%3Asourcegraph

cc @gregmagolan 

I expected to see a `storybook_test` or similar bazel rule. Something like jest_test https://docs.aspect.build/rules/aspect_rules_jest/docs/jest_test/

**Pull request merged**